client/packages/invoices/src/OutboundShipment/DetailView/DetailView.tsx

```diff
@@ -92,181 +87,47 @@ const DetailViewInner = () => {
-        filterVariant: 'select',
-        defaultHideOnMobile: true,
-      },
-      {
-        accessorKey: 'packSize',
-        header: t('label.pack-size'),
-        columnType: ColumnType.Number,
-        defaultHideOnMobile: true,
-      },
-      {
-        id: 'itemDoses',
-        header: t('label.doses-per-unit'),
-        columnType: ColumnType.Number,
-        defaultHideOnMobile: true,
-        accessorFn: row => (row.item.isVaccine ? row.item.doses : undefined),
-      },
-      {
-        accessorKey: 'numberOfPacks',
-        header: t('label.pack-quantity'),
-        columnType: ColumnType.Number,
-        accessorFn: row => {
-          if ('subRows' in row)
-            return ArrayUtils.getSum(row.subRows ?? [], 'numberOfPacks');
-
-          return row.numberOfPacks;
-        },
-      },
-      {
-        id: 'unitQuantity',
-        header: t('label.unit-quantity'),
-        description: t('description.unit-quantity'),
-        columnType: ColumnType.Number,
-        defaultHideOnMobile: true,
-        accessorFn: row => {
-          if ('subRows' in row)
-            return ArrayUtils.getUnitQuantity(row.subRows ?? []);
-
-          return row.packSize * row.numberOfPacks;
-        },
-      },
-      {
-        id: 'doseQuantity',
-        header: t('label.doses'),
-        columnType: ColumnType.Number,
-        defaultHideOnMobile: true,
-        accessorFn: row => {
-          if (!row.item.isVaccine) return null;
-          if ('subRows' in row)
-            return (
-              ArrayUtils.getUnitQuantity(row.subRows ?? []) *
-              (row.item.doses ?? 1)
-            );
-
-          return row.packSize * row.numberOfPacks * (row.item.doses ?? 1);
-        },
-      },
-      {
-        id: 'unitSellPrice',
-        header: t('label.unit-sell-price'),
-        columnType: ColumnType.Currency,
-        defaultHideOnMobile: true,
-        accessorFn: rowData => {
-          if ('subRows' in rowData) {
-            return ArrayUtils.getAveragePrice(
-              rowData.subRows ?? [],
-              'sellPricePerPack'
-            );
-          } else {
-            if (isDefaultPlaceholderRow(rowData)) return undefined;
-            return (rowData.sellPricePerPack ?? 0) / rowData.packSize;
-          }
-        },
-      },
-      {
-        id: 'total',
-        header: t('label.total'),
-        columnType: ColumnType.Currency,
-        defaultHideOnMobile: true,
-        accessorFn: rowData => {
-          if ('subRows' in rowData) {
-            return Object.values(rowData.subRows ?? []).reduce(
-              (sum, batch) =>
-                sum + batch.sellPricePerPack * batch.numberOfPacks,
-              0
-            );
-          } else {
-            if (isDefaultPlaceholderRow(rowData)) return '';
-
-            const x = rowData.sellPricePerPack * rowData.numberOfPacks;
-            return x;
-          }
-        },
-      },
-    ];
+  const columns = useOutboundColumns();
```

Comment:
This is quite a long column definition, so moved into hook, replacing the old useOutboundColumns 🙏